### PR TITLE
Fix 2388 obelisks puzzle revealing

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Teams and players were messed up in lib; hardcoded constants were refactored.

http://bugs.vcmi.eu/view.php?id=2388

### CODE REVIEW

File: lib/mapObjects/MiscObjects.h [hunk 1]
```diff
@@ -451,6 +453,7 @@ class DLL_LINKAGE CGDenOfthieves : public CGObjectInstance
 class DLL_LINKAGE CGObelisk : public CPlayersVisited
 {
 public:
+	static constexpr int OBJPROP_INC = 20;
```
Comment (DjWarmonger) [inline lib/mapObjects/MiscObjects.h]: What does "INC" stand for?

**Comment by DjWarmonger** [comment]
So many changes in blank lines make the commit difficult to read.

**Comment by vmarkovtsev** [comment]
I can split it into 2 separate commits: actual fix and whitespaces. But do I have to?

**Comment by vmarkovtsev** [comment]
See, many files contain trailing whitespaces and they suck.

File: lib/mapObjects/MiscObjects.h (hunk 1 above)
Comment (vmarkovtsev) [inline lib/mapObjects/MiscObjects.h]: Increment (that's the obelisk logic behind "20") - look at line 1847

**Comment by vmarkovtsev** [comment]
I found a couple of other bugs besides. Attached to 2388.

**Comment by vmarkovtsev** [comment]
Any issue with these?

**Comment by ArseniyShestakov** [comment]
I'll merge it once I test it appropriately (multiplayer, different maps).

**Comment by ArseniyShestakov** [comment]
Tested and compared with H3. We have slightly difference in how we reveal map, but no real reason to care about it no way game breaking. May put some suggestion later on on bug tracker, but for now I'll merge this one.

**Pull request merged**